src/plot_api/plot_api.js

```diff
@@ -302,7 +303,7 @@ Plotly.plot = function(gd, data, layout, config) {
                     .selectAll(query)
                     .remove();
 
-                fullLayout._infolayer.selectAll('g.rangeslider-container')
+                rangesliderContainers
```

Comment:
Right. I understand the value of caching selections, but here you're just assigning the selection to a variable: `.selectAll` is still called inside a loop.